src/controller/CHIPDeviceController.cpp

```diff
@@ -109,7 +109,7 @@ DeviceController::DeviceController()
     mExchangeMgr              = nullptr;
     mStorageDelegate          = nullptr;
     mPairedDevicesInitialized = false;
-    mListenPort               = CHIP_PORT;
+    mListenPort               = 0;
```

Comment:
What about setting the port, rather than letting it be random? It would definitely help test cases to allow setting externally, before starting the listen in Init. While the default value may be zero, a setter called before init, or a new param in ControllerInitParams is required to make this follow usual convention of "a server can choose its port if it desires.

Reply:
I'll push an update to this PR that adds the ListenPort to the init params.